windows_service/datadog_checks/windows_service/windows_service.py

```diff
@@ -1,101 +1,82 @@
-        tags.extend(custom_tags)
-
-        if len(services) == 0:
-            raise Exception('No services defined in windows_service.yaml')
-
-        properties = ["Name", "State"]
-        if "ALL" in services:
-            self.log.debug("tracking all services")
-            filters = None
-        else:
-            filters = map(lambda x: {"Name": tuple(('LIKE', x)) if '%' in x else tuple(('=', x))}, services)
-
-        wmi_sampler = self._get_wmi_sampler(
-            instance_key,
-            self.CLASS, properties,
-            filters=filters,
-            host=host, namespace=self.NAMESPACE,
-            username=user, password=password
-        )
+        if not services:
+            raise ValueError('No services defined in configuration.')
+
+        # Old-style WMI wildcards
+        if 'host' in instance:
+            services = set(service.replace('%', '.*') for service in services)
+
+        service_patterns = {service: re.compile(service) for service in services}
+        services_unseen = set(services)
 
         try:
-            # Sample, extract & submit metrics
-            wmi_sampler.sample()
-        except TimeoutException:
-            self.log.warning(
-                u"[WinService] WMI query timed out."
-                u" class={wmi_class} - properties={wmi_properties} -"
-                u" filters={filters} - tags={tags}".format(
-                    wmi_class=self.CLASS, wmi_properties=properties,
-                    filters=filters, tags=tags
-                )
-            )
-        else:
-            self._process_services(wmi_sampler, services, tags)
-
-    def _process_services(self, wmi_sampler, services, tags):
-        specific_services = {}
-        for svc in services:
-            if svc == "ALL":
-                continue
-            if '%' in svc:
-                continue
-            specific_services[svc.lower()] = svc
-
-        for wmi_obj in wmi_sampler:
-            sc_name = wmi_obj['Name']
-            sc_name_lower = sc_name.lower()
-            if sc_name_lower in specific_services:
-                try:
-                    specific_services.pop(sc_name_lower, None)
-                except KeyError:
-                    pass
-
-            status = self.STATE_TO_VALUE.get(wmi_obj["state"], WinWMICheck.UNKNOWN)
-            self.service_check("windows_service.state", status, tags=tags + ['service:{}'.format(sc_name)])
-            self.log.debug("service state for %s %s" % (sc_name, str(status)))
-
-        for lsvc, svc in specific_services.items():
-            self.service_check("windows_service.state", WinWMICheck.CRITICAL, tags=tags + ['service:{}'.format(svc)])
-            self.log.debug("service state for %s %s" % (svc, str(WinWMICheck.CRITICAL)))
+            scm_handle = win32service.OpenSCManager(None, None, win32service.SC_MANAGER_ENUMERATE_SERVICE)
+        except Exception as e:  # no cov
+            raise Exception('Unable to open SCManager: {}'.format(e))
```

Comment:
this should be OK, but we should explicitly try on a `dd-agent-user` build

Reply:
Just tested, it works!